packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -66,10 +72,30 @@ export class AndroidEmulatorDevice extends DeviceBase {
     ]);
   }
 
+  private async ensureOldEmulatorProcessExited() {
+    return new Promise<void>((resolve, reject) => {
+      let runningPid: string | undefined;
+      const subprocess = exec("ps", ["-A"]);
+      const regexpPattern = new RegExp(`(\\d+)\\s.*qemu.*-avd ${this.avdId}`);
+      lineReader(subprocess).onLineRead(async (line) => {
+        const regExpResult = regexpPattern.exec(line);
+        if (regExpResult) {
+          runningPid = regExpResult[1];
+        }
+      });
+
+      subprocess.on("close", async () => {
+        if (runningPid) {
+          await exec("kill", ["-9", `${runningPid}`]);
+        }
+        resolve();
+      });
+    });
```

Comment:
1) use `-o` to format ps output (it may be different depending on system settings)
2) Don't use promise, just await on subprocess and then do `if (runingPid) await exec(...)`

Reply:
done